ISSUE 41 # update de.tudarmstadt.ukp.wikipedia version numbers

sparrowV opened the pull request:
I took a look at maven repository and also https://oss.sonatype.org/service/local/staging/deploy/maven2/ repository and found out that there was no 1.2.0-SNAPSHOT version number which I could find of. 1.1.0 version number was present in mutiple places so I changed  it to that version. Warnings now are gone. I also ran test and everything is working fine.

sparrowV commented:
@tgalery does this project have username/password for https://oss.sonatype.org/ nexus repository?
when I try to go to https://oss.sonatype.org/service/local/staging/deploy/maven2/ ( where maven tries to download 1.2.0-SNAPSHOT version) it asks me to log in and I think thats the reason why we get maven warning not authorised

sparrowV commented:
@tgalery  also when I go to https://oss.sonatype.org/ , and in search bar I specify de.tudarmstadt.ukp.wikipedia.timemachine, I dont get 1.2.0-SNAPSHOT in the list, why is that?

tgalery commented:
My feeling is that the snapshots get published somewhere else. You can check the dkpro repo but my point is that the project is still buildable as is, right ?

sparrowV commented:
@tgalery   I fixed it! I changed the oss.sonatype.org repository for snapshots and now previous warning are gone, project builds sucessfully and all tests passes. I think we can merge this.

sparrowV commented:
@tgalery  I kept both repositories. Im not sure about squishing,since I pushed them. In gitlab there is a check box 'squish commit' when you are merging the branch, maybe github also has the same feature when you are about to merge?

sparrowV commented:
@tgalery  check out this  https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges , when you are about to merge you should have the following option Squash and merge

diegoceccarelli commented:
Thanks @sparrowV, I'll let @tgalery review and merge. I set squash and merge as the only way to merge the PR so you don't have to worry about squashing before. 

@tgalery I recalled that there was some reason why we needed `1.2.0-SNAPSHOT`, but i couldn't remember it - should we have a test for japanese? so that it fails if we use the wrong version?

tgalery commented:
@sparrowV Yeah I can squash your commits within a merge. For future reference it might be good to first open an issue so we can discuss and then have prs that solve that issue. AFAIK the issue would be:
* the `dkpro` lib displays some warnings upon build, right ?
I've checked out your branch and it seems that I still get the same warnings.
Moreover, both master branches (from here and your fork) seem to failing for me when I run `mvn compile` or `mvn assembly:assembly` with 
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.3:compile (default-compile) on project json-wikipedia: Fatal error compiling: invalid target release: 11 -> [Help 1]
```

sparrowV commented:
@tgalery  the issue was opened by diego ( https://github.com/diegoceccarelli/json-wikipedia/issues/41) which my pull request corresponds to.
With regard to the error, I also had same error but in my case it was not related to the project.
It was because I did not have JAVA_HOME variable point to jdk 11 ( maven looks for this variable). 
Can you make JAVA_HOME  variable point to 11 and try again?

tgalery commented:
Sorry about that @sparrowV it's been a long time since I done java
so for be it builds now, but I'm still having warnings like:

```
[WARNING] Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.revisionmachine:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Failure to transfer de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.revisionmachine:1.2.0-SNAPSHOT/maven-metadata.xml from https://oss.sonatype.org/service/local/staging/deploy/maven2/ was cached in the local repository, resolution will not be reattempted until the update interval of ossrh has elapsed or updates are forced. Original error: Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.revisionmachine:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
Downloading from de.tudarmstadt.ukp.wikipedia: http://zoidberg.ukp.informatik.tu-darmstadt.de/artifactory/public-snapshots/de/tudarmstadt/ukp/wikipedia/de.tudarmstadt.ukp.wikipedia.timemachine/1.2.0-SNAPSHOT/maven-metadata.xml
Downloading from ossrh: https://oss.sonatype.org/service/local/staging/deploy/maven2/de/tudarmstadt/ukp/wikipedia/de.tudarmstadt.ukp.wikipedia.timemachine/1.2.0-SNAPSHOT/maven-metadata.xml
Downloaded from de.tudarmstadt.ukp.wikipedia: http://zoidberg.ukp.informatik.tu-darmstadt.de/artifactory/public-snapshots/de/tudarmstadt/ukp/wikipedia/de.tudarmstadt.ukp.wikipedia.timemachine/1.2.0-SNAPSHOT/maven-metadata.xml (1.5 kB at 15 kB/s)
[WARNING] Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.timemachine:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Failure to transfer de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.timemachine:1.2.0-SNAPSHOT/maven-metadata.xml from https://oss.sonatype.org/service/local/staging/deploy/maven2/ was cached in the local repository, resolution will not be reattempted until the update interval of ossrh has elapsed or updates are forced. Original error: Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.timemachine:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
```
which is what you meant to fix ?

sparrowV commented:
@tgalery  it should not  give you wanrning about de.tudarmstadt.ukp.wikipedia failure of tranfer matadata. can you do `maven clean compile` and check if that warnings are gone?

tgalery commented:
```
$ git log
commit 137251372439781099ad8005a7d0f702d3d688ac (HEAD -> a-master, sparrow/master)
Author: Otto <otto.dvali@gmail.com>
Date:   Mon May 4 20:32:20 2020 +0400

    minor refactor
$ mvn clean compile | grep -i warning
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[WARNING] Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for it.cnr.isti.hpc:json-wikipedia:jar:2.0.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.(groupId:artifactId)' must be unique but found duplicate declaration of plugin org.apache.maven.plugins:maven-compiler-plugin @ line 216, column 29
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-surefire-plugin is missing. @ line 209, column 14
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[WARNING] Could not transfer metadata info.bliki.wiki:bliki-core:3.1.1-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Failure to transfer info.bliki.wiki:bliki-core:3.1.1-SNAPSHOT/maven-metadata.xml from https://oss.sonatype.org/service/local/staging/deploy/maven2/ was cached in the local repository, resolution will not be reattempted until the update interval of ossrh has elapsed or updates are forced. Original error: Could not transfer metadata info.bliki.wiki:bliki-core:3.1.1-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Could not transfer metadata info.bliki.wiki:bliki:3.1.1-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.api:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Failure to transfer de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.api:1.2.0-SNAPSHOT/maven-metadata.xml from https://oss.sonatype.org/service/local/staging/deploy/maven2/ was cached in the local repository, resolution will not be reattempted until the update interval of ossrh has elapsed or updates are forced. Original error: Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.api:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Failure to transfer de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia:1.2.0-SNAPSHOT/maven-metadata.xml from https://oss.sonatype.org/service/local/staging/deploy/maven2/ was cached in the local repository, resolution will not be reattempted until the update interval of ossrh has elapsed or updates are forced. Original error: Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.parser:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Failure to transfer de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.parser:1.2.0-SNAPSHOT/maven-metadata.xml from https://oss.sonatype.org/service/local/staging/deploy/maven2/ was cached in the local repository, resolution will not be reattempted until the update interval of ossrh has elapsed or updates are forced. Original error: Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.parser:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.datamachine:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Failure to transfer de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.datamachine:1.2.0-SNAPSHOT/maven-metadata.xml from https://oss.sonatype.org/service/local/staging/deploy/maven2/ was cached in the local repository, resolution will not be reattempted until the update interval of ossrh has elapsed or updates are forced. Original error: Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.datamachine:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.wikimachine:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Failure to transfer de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.wikimachine:1.2.0-SNAPSHOT/maven-metadata.xml from https://oss.sonatype.org/service/local/staging/deploy/maven2/ was cached in the local repository, resolution will not be reattempted until the update interval of ossrh has elapsed or updates are forced. Original error: Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.wikimachine:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.mwdumper:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Failure to transfer de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.mwdumper:1.2.0-SNAPSHOT/maven-metadata.xml from https://oss.sonatype.org/service/local/staging/deploy/maven2/ was cached in the local repository, resolution will not be reattempted until the update interval of ossrh has elapsed or updates are forced. Original error: Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.mwdumper:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.revisionmachine:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Failure to transfer de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.revisionmachine:1.2.0-SNAPSHOT/maven-metadata.xml from https://oss.sonatype.org/service/local/staging/deploy/maven2/ was cached in the local repository, resolution will not be reattempted until the update interval of ossrh has elapsed or updates are forced. Original error: Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.revisionmachine:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.timemachine:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Failure to transfer de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.timemachine:1.2.0-SNAPSHOT/maven-metadata.xml from https://oss.sonatype.org/service/local/staging/deploy/maven2/ was cached in the local repository, resolution will not be reattempted until the update interval of ossrh has elapsed or updates are forced. Original error: Could not transfer metadata de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.timemachine:1.2.0-SNAPSHOT/maven-metadata.xml from/to ossrh (https://oss.sonatype.org/service/local/staging/deploy/maven2/): Not authorized , ReasonPhrase:Unauthorized.
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
```
This is what I get from your branch

sparrowV commented:
@tgalery  can you try now?

tgalery commented:
Tried and it's working legit without the warning. Merging it now. Thanks for the work!
